src/components/PermissionsModal/PermissionsModal.js

```diff
@@ -0,0 +1,322 @@
+import {
+  isEmpty,
+  get,
+  orderBy,
+  remove,
+} from 'lodash';
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect as reduxConnect } from 'react-redux';
+import { FormattedMessage } from 'react-intl';
+import { change } from 'redux-form';
+
+import { stripesConnect } from '@folio/stripes/core';
+import {
+  Button,
+  Modal,
+  Pane,
+  Paneset,
+  ModalFooter,
+} from '@folio/stripes/components';
+
+import SearchForm from './components/SearchForm';
+import PermissionsList from './components/PermissionsList';
+import { sortOrders } from './constants';
+import { getInitialFiltersState } from './helpers';
+
+import css from './PermissionsModal.css';
+
+class PermissionsModal extends React.Component {
+  static manifest = Object.freeze({
+    availablePermissions: {
+      type: 'okapi',
+      records: 'permissions',
+      path: 'perms/permissions?length=10000&query=(mutable==false and visible==true)',
+      fetch: false,
+      accumulate: true,
+    },
+  });
+
+  static propTypes = {
+    mutator: PropTypes.shape({
+      availablePermissions: PropTypes.shape({
+        GET: PropTypes.func.isRequired,
+        reset: PropTypes.func.isRequired,
+        POST: PropTypes.func.isRequired,
+      }),
+    }),
+    resources: PropTypes.shape({
+      availablePermissions: PropTypes.shape({
+        records: PropTypes.arrayOf(PropTypes.object),
+      }),
+    }),
+    filtersConfig: PropTypes.arrayOf(PropTypes.object).isRequired,
```

Comment:
Could you please use PropTypes.shape instead of PropTypes.object. Do the same for subPermissions property also.